File: C#/107_BinaryTreeLevelOrderTraversalII.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrderBottom(TreeNode root) {
        if (root == null) return new List<IList<int>>();
        
        Stack<IList<int>> ans = new Stack<IList<int>>();       
        Queue<(int,TreeNode)> q = new Queue<(int, TreeNode)>();
        
        q.Enqueue((0,root));
        while (q.Count > 0) {
            (int h, TreeNode node) = q.Dequeue();
            IList<int> lst = ans.Count > h? ans.Pop() : new List<int>(); 
            lst.Add(node.val);
            ans.Push(lst);
            if (node.left != null) q.Enqueue((h + 1, node.left));
            if (node.right != null) q.Enqueue((h + 1, node.right));
        }
        
        return ans.ToList();
    }
}

File: Python/107_BinaryTreeLevelOrderTraversalII.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        if root is None: return []
        q = deque()
        q.append([0,root])
        ans = []
        while q:
            h,n = q.popleft()
            if len(ans) <= h: ans.append([])
            ans[h].append(n.val)           
            if n.left != None: q.append([h + 1, n.left])
            if n.right != None: q.append([h + 1, n.right])
            
        return reversed(ans)
    
    